lib/charms/tempo_k8s/v1/charm_tracing.py

```diff
@@ -15,14 +15,17 @@
 `@trace_charm(tracing_endpoint="my_tracing_endpoint")`
 
 2) add to your charm a "my_tracing_endpoint" (you can name this attribute whatever you like) **property**
-that returns an otlp grpc endpoint url. If you are using the `TracingEndpointProvider` as
+that returns an otlp http endpoint url. If you are using the `TracingEndpointProvider` as
 `self.tracing = TracingEndpointProvider(self)`, the implementation could be:
 
 ```
     @property
     def my_tracing_endpoint(self) -> Optional[str]:
         '''Tempo endpoint for charm tracing'''
-        return self.tracing.otlp_grpc_endpoint
+        if self.tracing.is_ready():
+            return self.tracing.otlp_http_endpoint()
+        else:
+            return None
 ```
```

Comment:
It's not standard practice (yet, but why not), but I think it would be a good idea to add in this docstring an "upgrading from v0" section giving a minimal example of what you need to change in charm code to port a charm to use the new lib.
No need to mention charmcraft fetch-lib, just focus on 'user code'.

Not a hard requirement for this PR, just a thought if you like the idea. Otherwise we can document it elsewhere.

Reply:
Nice idea, I added a short `upgrading from v0` section at the end of this docstring.